chips/earlgrey/src/chip.rs

```diff
@@ -212,28 +212,25 @@ impl<'a, I: InterruptService<()> + 'a> kernel::platform::chip::Chip for EarlGrey
 
     fn service_pending_interrupts(&self) {
         loop {
-            let mip = CSR.mip.extract();
-
             if self.plic.get_saved_interrupts().is_some() {
                 unsafe {
                     self.handle_plic_interrupts();
                 }
             }
 
-            if !mip.matches_any(mip::mtimer::SET) && self.plic.get_saved_interrupts().is_none() {
+            if self.plic.get_saved_interrupts().is_none() {
                 break;
             }
         }
 
         // Re-enable all MIE interrupts that we care about. Since we looped
         // until we handled them all, we can re-enable all of them.
-        CSR.mie.modify(mie::mext::SET + mie::mtimer::SET);
+        CSR.mie.modify(mie::mext::SET + mie::mtimer::CLEAR);
```

Comment:
```suggestion
        CSR.mie.modify(mie::mext::SET);
```

Given that mtimer should already be cleared, isn't this better?

Reply:
It should already be cleared, that's true. I like this though as it makes it explicit that we aren't using the standard RISC-V timer interrupt. This is different to every other RISC-V chip so I think it's worth making it obvious here